src/compiler/checker.ts

```diff
@@ -888,8 +888,12 @@ namespace ts {
 
         function getTargetOfImportClause(node: ImportClause): Symbol {
             const moduleSymbol = resolveExternalModuleName(node, (<ImportDeclaration>node.parent).moduleSpecifier);
+
             if (moduleSymbol) {
-                const exportDefaultSymbol = resolveSymbol(moduleSymbol.exports["default"]);
+                const exportDefaultSymbol = moduleSymbol.exports["export="] ?
+                    getPropertyOfType(getTypeOfSymbol(moduleSymbol.exports["export="]), "default") :

```

Comment:
That does indeed seem to work for `import val from './mod'` syntax. Interesting that `import * as m from './mod'` doesn't allow the type of assignment you write above.

What is the expected behavior with `export = <expr>`? The spec (section 11.3.5) still states the grammar is `export = <IdentifierReference>;`, so I was expecting a parse error (but apparently we changed this?)